sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1093,6 +1131,22 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.23
 
+    interaction_cst : iterable of iterables of int, default=None
```

Comment:
> The resulting trees are not identical, but the resulting model structure is in the sense that the interaction constraints are fulfilled in both cases.

The constraint is fulfilled but the models are not equivalent in any way, right? Aka what @ogrisel  said, it's quite a different model.

> But the tree will usually use three features.

One of the reasons people restrict to pairwise interactions is so that the full model can be visualized. That's much harder with three features. There is no way to achieve trees that are on pairs of features with this PR, right?